### ISSUE TITLE
Changing between languages keeps results

### ISSUE THREAD

**Issue by zb-markus** [klokantech/kartenportal-tickets#128]
When the user decides to change the language, for example from English to German, his results should be kept. That is, the position of the search map, the zoom level, and therefore the order of the Trefferliste should not get lost. If a Detailansicht of a record is open, a change of the language should be possible without loosing the Detailansicht of that record.

**Comment by klokan** [comment]
Done and deployed.

**Comment by zb-markus** [comment]
While the change of language now keeps the Detailansicht, the order within the Trefferliste is not kept and the highlighted record is no longer highlighted. This is quite confusing for the user. Please:
- Keep the order of records within Trefferliste (which seems to be feasible because nothing else changes when only the language button is used).
- The selected record remains highlighted within Trefferliste.
- The highlighted record within the Trefferliste remains visible on the screen even when the scrollbar had been used, or in other words, the previously selected position of the Trefferliste is kept even when the selected record is from further down the list.
  Do you think that these 3 issues can be solved from the technical point of view?

**Comment by klokan** [comment]
Marcus, please keep in mind that switching of the language is extremely rarely used function for normal visitors. Practically all the users do that only once or never at all.

What you are requesting here is doable with a lot of extra work - and it means to significantly change the implementation of permalinks and storing and loading of the context of the application.

Kartenportal Search is an AJAX application. Localized versions run on different domains so if you switch the language you are loading a different website and starting a completely different web application. All the application context is lost. We have available only the context mentioned in the permalink - and there is only the identifier of the actual record for the opened detail window.
This allows to keep the permalink requested in the F.20 short, neat and readable.

Otherwise we would need to store many more status information, and as a result the permalink would be very long (not only what record we see, but also what is exactly the query which leads to the selected record and what are all the parameters, filters on the library, filters on time, etc. - and as an extra also how far are you scrolled and what is exactly the page on which actual record is present). 

!!!  Work on something like this was not estimated, and it is not part of the specification of the project, neither in F.20 (Detailansicht) nor F.35 (Localisierung).

There is in fact no special section dedicated to permalink in the Lastenheft. Despite that we have deployed perfectly running permalinks outside of the "Detailansichts" window.
BTW The Gmail interface reloads completely as well - if the language is changed.

**Comment by klokan** [comment]
To be discussed tomorrow.

**Comment by zb-markus** [comment]
Tested and accepted.
